src/dune_engine/static_deps.ml

```diff
@@ -21,3 +21,9 @@ let union x y =
 
 let paths { action_deps; rule_deps } =
   Dep.Set.paths (Dep.Set.union action_deps rule_deps)
+
+let union_map xs ~f =
+  let sets = List.map xs ~f in
+  { rule_deps = Dep.Set.union_map sets ~f:(fun x -> x.rule_deps)
+  ; action_deps = Dep.Set.union_map sets ~f:(fun x -> x.action_deps)
```

Comment:
It seems a bit strange to compute the union twice, discarding half of the output each time. Wouldn't it be faster to do it only once, followed by two `map`s?

Reply:
Hmm, but we're not computing the union twice. The only computation we're repeating is folding over `sets` twice. Which I suppose we could avoid by folding. I'll do that.